var/spack/repos/builtin/packages/py-celery/package.py

```diff
@@ -65,10 +67,29 @@ class PyCelery(PythonPackage):
     depends_on("py-click-repl@0.2.0:", when="@5.2.0:", type=("build", "run"))
     depends_on("py-pytz@2019.3:", type=("build", "run"))
     depends_on("py-pytz@2021.3:", type=("build", "run"), when="@5.2.3")
-    depends_on("py-billiard@3.6.3.0:3", type=("build", "run"))
+    depends_on("py-billiard@3.6.3.0:3", type=("build", "run"), when="@:5.0.0")
     depends_on("py-billiard@3.6.4.0:3", type=("build", "run"), when="@5.2.3")
     depends_on("py-kombu@4.6.11", when="@4.3.0:4", type=("build", "run"))
     depends_on("py-kombu@5.0.0:", when="@5.0.0:5.0", type=("build", "run"))
     depends_on("py-kombu@5.2.3:5", when="@5.2.0:5.2", type=("build", "run"))
     depends_on("py-vine@1.3.0", when="@4.3.0:4", type=("build", "run"))
     depends_on("py-vine@5.0.0:5", when="@5.0.0:5", type=("build", "run"))
+
+    with when("@5.2.3:5"):
+        depends_on("py-cryptography", type=("build", "run"))
+
+    with when("@5.3.4:5"):
+        depends_on("py-billiard@4.1:4", type=("build", "run"))
+        depends_on("py-kombu@5.3.2:5", type=("build", "run"))
+        depends_on("py-vine@5.0.0:5", type=("build", "run"))
+        depends_on("py-click@8.1.2:8", type=("build", "run"))
+        depends_on("py-click-didyoumean@0.3.0:", type=("build", "run"))
+        depends_on("py-click-repl@0.2.0:", type=("build", "run"))
+        depends_on("py-click-plugins@1.1.1:", type=("build", "run"))
```

Comment:
This one I skipped consciously, since the package also says it supports Python 3.8 and higher, see:

https://github.com/celery/celery/blob/0bc89cc594638e1d88655764807fcc59fb32efc6/setup.py#L172-L176